library/src/main/java/com/ironz/binaryprefs/BinaryPreferencesBuilder.java

```diff
@@ -0,0 +1,95 @@
+import com.ironz.binaryprefs.task.ScheduledBackgroundTaskExecutor;
+import com.ironz.binaryprefs.task.TaskExecutor;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BinaryPreferencesBuilder {
+
+    private static final String DEFAULT_NAME = "default";
+    private final Context context;
+    private final Map<String, Class<? extends Persistable>> persistableByToken = new HashMap<>();
+    private byte[] secret;
+    private byte[] initialVector;
+    private String name = DEFAULT_NAME;
+    private ExceptionHandler exceptionHandler = ExceptionHandler.IGNORE;
+
+    public BinaryPreferencesBuilder(Context context) {
+        this.context = context;
+    }
+
+    public BinaryPreferencesBuilder encryption(byte[] secret, byte[] initialVector) {
+        this.secret = secret;
+        this.initialVector = initialVector;
+        return this;
+    }
+
+    public BinaryPreferencesBuilder exceptionHandler(ExceptionHandler exceptionHandler) {
+        this.exceptionHandler = exceptionHandler;
+        return this;
+    }
+
+    public BinaryPreferencesBuilder name(String name) {
+        this.name = name;
+        return this;
+    }
+
+    public BinaryPreferencesBuilder registerPersistableToken(String token, Class<? extends Persistable> persistable) {
+        persistableByToken.put(token, persistable);
+        return this;
+    }
+
+    public BinaryPreferencesBuilder registerPersistableTokens(Map<String, Class<? extends Persistable>> persistableByToken) {
+        this.persistableByToken.putAll(persistableByToken);
+        return this;
+    }
+
+    public final Preferences build() {
+        DirectoryProvider directoryProvider = new AndroidDirectoryProviderImpl(context, name);
+        FileAdapter fileAdapter = new NioFileAdapter(directoryProvider);
+        LockFactory lockFactory = new SimpleLockFactoryImpl(name, directoryProvider);
+        FileTransaction fileTransaction = new MultiProcessTransactionImpl(fileAdapter, lockFactory);
+        ByteEncryption byteEncryption = getByteEncryption();
+        CacheProvider cacheProvider = new ConcurrentCacheProviderImpl(name);
+        TaskExecutor executor = new ScheduledBackgroundTaskExecutor(exceptionHandler);
+        PersistableRegistry persistableRegistry = new PersistableRegistry();
+        if (!persistableByToken.isEmpty()) {
+            for (Map.Entry<String, Class<? extends Persistable>> entry : persistableByToken.entrySet()) {
+                persistableRegistry.register(entry.getKey(), entry.getValue());
+            }
+        }
+        SerializerFactory serializerFactory = new SerializerFactory(persistableRegistry);
+        EventBridge eventsBridge = new BroadcastEventBridgeImpl(context, name, cacheProvider, serializerFactory, executor, byteEncryption);
+        return new BinaryPreferences(fileTransaction, byteEncryption, eventsBridge, cacheProvider, executor, serializerFactory, lockFactory);
+    }
+
+    private ByteEncryption getByteEncryption() {
```

Comment:
I think you can just create public builder api method `byteEncryption` method which should take `ByteEncryption` instance. And in `build method you shuld check if `ByteEncryption` is not defined - you will throw exception.